### PR TITLE
rename notification to event

### PR DESCRIPTION
**Pull request by kattrali**
Small change (smaller than it looks!) to bring bugsnag-python in line with other libraries' nomenclature.

* Renames `Notification` class to `Event`
* Adds new `Notification` shim class which warns if instantiated (but otherwise functions normally)
* Replaces existing use of the `Notification` class with `Event`

### CODE REVIEW

**Review by imjoehaines — COMMENTED** [review]
There are still a lot of `notification`s in the code; do we want to do anything about them too?

I think the functions (like `add_wsgi_request_data_to_notification`) should be safe to rename as they're internal

The parameters could be a BC break if they're called as a named parameter, but I'm not sure how common that'd be in practice, so maybe they're OK to rename too?

File: tests/test_notification.py [hunk 1]
```diff
@@ -1,184 +1,15 @@
...
-        It should include app type if specified
-        """
-        config = Configuration()
-        config.configure(app_type='rq')
-        notification = Notification(Exception("oops"), config, {})
-        payload = json.loads(notification._payload())
-        app = payload['events'][0]['app']
-
-        assert app['type'] == 'rq'
+def test_create_notification_warns():
```
[the remark this answers is not in the document]
Comment (kattrali) [reply tests/test_notification.py:7]: That's a good point - I realized there's a low overhead way to do this too.

**Comment by kattrali** [comment]
> There are still a lot of `notification`s in the code; do we want to do anything about them too?

I was somewhat torn on these, in case it was breaking to change parameter names, though since it also seems unlikely to you that it would be an issue, I went ahead and updated everything aside from the integration function names (variants of `add_{blah}_to_notification`).

**Comment by imjoehaines** [comment]
> > There are still a lot of `notification`s in the code; do we want to do anything about them too?
> 
> I was somewhat torn on these, in case it was breaking to change parameter names, though since it also seems unlikely to you that it would be an issue, I went ahead and updated everything aside from the integration function names (variants of `add_{blah}_to_notification`).

We don't document those functions AFAICT and I think they're unlikely to be used directly by users as they're applied by the respective middleware. I don't think it's a huge deal either way though; we can always rename them when we remove `Notification` entirely

I suppose we could even have `add_{blah}_to_event` versions as duplicates, but that's probably overkill

**Pull request merged**